subiquity/server/controllers/source.py

```diff
@@ -148,10 +148,30 @@ def get_handler(
 
     async def configured(self):
         await super().configured()
+        self._configured = True
         self.app.base_model.set_source_variant(self.model.current.variant)
 
     async def POST(self, source_id: str, search_drivers: bool = False) -> None:
-        self.model.search_drivers = search_drivers
-        with contextlib.suppress(KeyError):
-            self.model.current = self.model.get_matching_source(source_id)
-        await self.configured()
+        # Marking the source model configured has an effect on many of the
+        # other controllers. Oftentimes, it would involve cancelling and
+        # restarting various operations.
+        # Let's try not to trigger the event again if we are not changing any
+        # of the settings.
+        changed = False
+        if self.model.search_drivers != search_drivers:
+            changed = True
+            self.model.search_drivers = search_drivers
+
+        try:
+            new_source = self.model.get_matching_source(source_id)
+        except KeyError:
+            # TODO going forward, we should probably stop silently ignoring
```

Comment:
Yes, that's a good point, I don't like the silent ignore.  At minimum let's log.

This scenario is not an autoinstall one - that's handled above with `load_autoinstall_data()` / `start()`, so I think the only cases where this could happen are
* client bug
* server bug
* hardcoded source value in an automated test.

I suspect we should log and `raise` here, what do you think?

Reply:
Yes I agree. I'm going to add the logging statement.

As for re-raising, IMO it depends if we feel like that this PR might make things better for mantic.  If we ponder landing this to ubuntu/mantic, I definitely don't want to re-raise. But going forward, I think we should!

IIRC, there might be a race condition in some scenarios between:
* subiquity reading the source catalog
* receiving a POST to /source.

Reply:
ah yes, I'm presuming the read of the source catalog has happened here.
OK.  Let's go with the error level log for now.